src/Blocks/custom/carousel/assets/carousel-slider.js

```diff
@@ -0,0 +1,65 @@
+import Swiper, { Navigation, Pagination } from 'swiper';
+
+Swiper.use([Navigation, Pagination]);
+
+export class CarouselSlider {
+	constructor(options) {
+		this.element = options.element;
+		this.blockClass = options.blockClass;
+		this.nextElement = options.nextElement;
+		this.prevElement = options.prevElement;
+		this.paginationElement = options.paginationElement;
+		this.eventName = options.eventName;
+		this.swipeHandler = options.eventName;
+	}
+
+	init() {
+		const item = this.element;
+
+		new Swiper(item, {
+			loop: item.getAttribute('data-swiper-loop'),
+			slideClass: `${this.blockClass}__item`,
```

Comment:
this selector must be passed from the constructor na not hardcoded in the class. Also all js selectors must have .js-prefix and we dont use bem in js selectors